src/server/api/routers/auth.test.ts

```diff
@@ -245,3 +246,56 @@ describe("auth.checkValidToken", () => {
     expect(result.success).toBe(true);
   });
 });
+
+describe("auth.resendEmail", () => {
+  test("unauthorized user: throws error", async () => {
+    const unauthCtx = createInnerTRPCContext({ session: null });
+    const unauthCaller = createCaller(unauthCtx);
+    await expect(unauthCaller.auth.resendEmail()).rejects.toThrowError(
+      "Not logged in",
+    );
+  });
+
+  test("non-existent user: throws error", async () => {
+    const fakeSession = {
+      ...session,
+      user: {
+        ...session.user,
+        id: "non-existent-id",
+        email: "nonexistent@example.com",
+      },
+    };
+    const fakeCtx = createInnerTRPCContext({ session: fakeSession });
+    const fakeCaller = createCaller(fakeCtx);
+    await expect(fakeCaller.auth.resendEmail()).rejects.toThrowError(
+      /not found/i,
+    );
+  });
+
+  test("existing user: returns success", async () => {
+    const sessionUser = session.user;
+    await db
+      .insert(users)
+      .values({
+        id: sessionUser.id,
+        name: sessionUser.name ?? faker.person.fullName(),
+        email: sessionUser.email ?? faker.internet.email(),
+        emailVerified: null,
+        image: sessionUser.image ?? faker.image.avatar(),
+      })
+      .onConflictDoNothing?.();
+
+    const sendEmailSpy = vi.spyOn(resend.emails, "send").mockResolvedValue({
+      data: { id: "mock-email-id" },
+      error: null,
+    });
+
+    try {
+      const result = await caller.auth.resendEmail();
+      expect(result.success).toBe(true);
+    } finally {
+      // Restore the original method with its context
+      sendEmailSpy.mockRestore();
+    }
```

Comment:
the `resend.emails` function already gets mocked, see `vitest.setup.ts` so I believe you can get away with just doing this

```suggestion
    const result = await caller.auth.resendEmail();
    expect(result.success).toBe(true);
    expect(resend.emails.send).toHaveBeenCalled(); 
```